catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeMapImpl.java

```diff
@@ -90,20 +94,30 @@ public String getProperty() {
     if (!isLeaf()) {
       throw new IllegalStateException("No property value exists for a logical operator");
     }
-    return (String) json.get(PROPERTY);
+    Object property = json.get(PROPERTY);
+    if (!(property instanceof String)) {
```

Comment:
same question as on https://github.com/codice/ddf/pull/6529/files#r589452378

Reply:
The `instanceof` check will return `false` if `property` is `null` and it'll only ever been `null` if the UI sends it that way.

Reply:
I use the alternate method, `getPropertyFunction` which doesn't fail that way, to check if it's a function. Otherwise we call this. If the UI does happen to send us a `null` node the whole request will fail gracefully since forms let exceptions propagate from the conversion up to the request level.